packages/react/src/primitives/shared/responsive/useBreakpoint.ts

```diff
@@ -0,0 +1,59 @@
+// Inspiration for getMediaQueries and useBreakpoint
+// comes from https://github.com/iiroj/use-breakpoint/
+
+import * as React from 'react';
+import { Breakpoint, UseBreakpoint } from '../../types/responsive';
+import { getMediaQueries } from './getMediaQueries';
+
+const useIsomorphicEffect =
+  typeof window === 'undefined' ? React.useEffect : React.useLayoutEffect;
+
+export const useBreakpoint: UseBreakpoint = ({
+  breakpoints,
+  breakpointUnit,
+  defaultBreakpoint,
+}) => {
+  const supportMatchMedia =
+    typeof window !== 'undefined' && typeof window.matchMedia !== 'undefined';
+  const matchMedia = supportMatchMedia ? window.matchMedia : null;
+
+  const mediaQueries = React.useMemo(
+    () => getMediaQueries({ breakpoints, breakpointUnit }),
+    [breakpoints, getMediaQueries]
+  );
+
+  const [breakpoint, setBreakpoint] =
+    React.useState<Breakpoint>(defaultBreakpoint);
+
+  const updateBreakpoint = React.useCallback(
+    (matches: boolean, breakpoint: Breakpoint) => {
+      if (matches) {
+        setBreakpoint(breakpoint);
+      }
+    },
+    [setBreakpoint]
+  );
+
+  useIsomorphicEffect(() => {
+    if (!matchMedia) return;
+    const unsubscribeList = mediaQueries.map(({ query, breakpoint }) => {
+      const queryList = matchMedia(query);
+      updateBreakpoint(queryList.matches, breakpoint);
+      const handleMediaChange = (event: MediaQueryListEvent) => {
+        if (event.matches) {
+          setBreakpoint(breakpoint);
+        }
+      };
+      queryList.addEventListener('change', handleMediaChange);
+      return () => queryList.removeEventListener('change', handleMediaChange);
+    });
+    return () => {
+      unsubscribeList.forEach((unsubscribe) => unsubscribe());
+    };
+  }, [breakpoints, setBreakpoint, mediaQueries]);
```

Comment:
Yeah, I only added it so that we could add the exhaustive-dependencies eslint rule in the near future without having to change this code.
https://www.npmjs.com/package/eslint-plugin-react-hooks

Reply:
That makes sense to me. But looks like one more dependency `matchMedia` is missing?

Reply:
Fixed, thanks for the catch!